### PR TITLE
Add the ability to pass env vars to repo run calls.

### PR DESCRIPTION
**Pull request by abayer**
Backwards compatible - old run API still available.

cc @reviewbybees esp @jglick

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/steps/scm/AbstractSampleDVCSRepoRule.java [hunk 1]
```diff
@@ -54,10 +56,14 @@ public final void write(String rel, String text) throws IOException {
     public abstract void init() throws Exception;
 
     protected final void run(String tool, String... cmds) throws Exception {
+        run(tool, null, cmds);
+    }
+
+    protected final void run(String tool, Map<String,String> env, String... cmds) throws Exception {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/steps/scm/AbstractSampleDVCSRepoRule.java]: :ant: add `@CheckForNull`

File: src/test/java/org/jenkinsci/plugins/workflow/steps/scm/AbstractSampleRepoRule.java [hunk 2]
```diff
@@ -37,13 +39,21 @@
 public abstract class AbstractSampleRepoRule extends ExternalResource {
 
     public static void run(boolean probing, File cwd, String... cmds) throws Exception {
+        run(probing, cwd, null, cmds);
+    }
+
+    public static void run(boolean probing, File cwd, Map<String,String> env, String... cmds) throws Exception {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/steps/scm/AbstractSampleRepoRule.java]: :ant: ditto

**Comment by jglick** [comment]
:bee: iff there is a downstream change which successfully uses this

**Comment by abayer** [comment]
Yup, wanted your thoughts on this first before I do a `git-plugin` PR (though I can't do deploys of `workflow-scm-step-plugin`, so the `git-plugin` PR will fail to build) and subsequent change to https://github.com/jenkinsci/workflow-job-plugin/pull/19.

**Comment by abayer** [comment]
Oh, bugger, `git-plugin` is still back depending on 1.14.2. Guess I'll have to fork `GitSampleRepoRule` in `workflow-job` for now?

**Comment by jglick** [comment]
> Guess I'll have to fork `GitSampleRepoRule` in `workflow-job` for now?

Mm, better not. I have a plan for various changes to be done in this code, which would include moving to `scm-api`, where `git` could pick it up without forking. Can you hold off on the test case that would use this? For example, use `Assume` to skip the test when the CI environment is found to be messed up?

**Comment by abayer** [comment]
Yeah, I can skip it when `GIT_COMMITTER_NAME` is set for now.

**Comment by jglick** [comment]
I think https://github.com/jenkinsci/git-plugin/pull/469 supersedes this.

**Pull request closed without merging**